scylla/src/transport/session.rs

```diff
@@ -1653,7 +1653,8 @@ impl Session {
         QueryFut: Future<Output = Result<ResT, QueryError>>,
         ResT: AllowedRunQueryResTType,
     {
-        let mut last_error: Option<QueryError> = None;
+        // set default error as no known found as the query plan returns an empty iterator if there are no nodes in the plan
+        let mut last_error: Option<QueryError> = Some(QueryError::NoKnownNodeFoundError("Please confirm the supplied datacenters exists".to_string()));
```

Comment:
The query_plan is an [iterator](https://github.com/scylladb/scylla-rust-driver/blob/main/scylla/src/transport/session.rs#L1661) that provides a Node, but this iterator can be empty due to the inability to [find/pick a node](https://github.com/scylladb/scylla-rust-driver/blob/main/scylla/src/transport/load_balancing/plan.rs#L71) which is mostly [caused here](https://github.com/scylladb/scylla-rust-driver/blob/main/scylla/src/transport/load_balancing/default.rs#L398). 

There could be other reasons why the iterator could be empty, but I can't see any yet